### PR TITLE
fix: polygon support

### PR DESCRIPTION
**Pull request by HJunyuan**
## Context
- `/email` and `/verify` does not work for Polygon-issued documents

## What does this PR do?
- Create a shared `verify()` to be used by the various lambda functions
- Auto switch between Ethereum and Polygon dependent on `document.network` field ([see reference implementation](https://github.com/OpenCerts/opencerts-website/blob/1de65c66795ec2f416d6c829259e9f9ab1e49e45/src/sagas/certificate.ts#L139))

### CODE REVIEW

File: src/shared/verify.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  return config.network;
+}
+
+/**
+ * A wrapper of verify to auto-switch between Ethereum and Polygon
+ */
+export const verify = (document) => {
+  if (!validateSchema(document)) {
+    // Following current behaviour of from "@govtechsg/opencerts-verify"
+    // E.g. await verify({ "bad": "document" }) // returns undefined
```
Comment (phanshiyu) [inline src/shared/verify.js:57]: weird is the return type for `verify` wrong? it should not return an `undefined` based on the type haha

File: src/storage/documentService/documentService.js [hunk 2]
```diff
@@ -68,7 +69,7 @@ const uploadDocumentAtId = async (
     throw new Error("Ttl cannot exceed 90 days");
   }
 
-  const fragments = await verify({ network: config.network })(document);
+  const fragments = await verify(document);
   if (!isValid(fragments)) {
```
Comment (phanshiyu) [inline src/storage/documentService/documentService.js:73]: if verify returns `undefined`, and hence `isValid(undefined)`, this error will occur, is that expected?:

```
  if (verificationFragments.length < 1) {
                            ^

TypeError: Cannot read properties of undefined (reading 'length')
```

File: src/shared/verify.js (hunk 1 above)
Comment (HJunyuan) [reply src/shared/verify.js]: I did some further digging and realised it wasn't `verify()` that was causing the issue I was facing originally.

 `registryVerifier` will not handle a bad document like `{ foo: "bar" }` correctly and throw this error:
```
TypeError: Cannot read properties of undefined (reading 'test')
```

File: src/storage/documentService/documentService.js (hunk 2 above)
Comment (HJunyuan) [reply src/storage/documentService/documentService.js:73]: [Refer to above comment]

File: src/shared/verify.js (hunk 1 above)
Comment (phanshiyu) [reply src/shared/verify.js]: i see!

**Review by phanshiyu — APPROVED** [review]

File: src/shared/verify.js (hunk 1 above)
Comment (HJunyuan) [reply src/shared/verify.js]: I've created a temporary branch (`temp/type-error`) to showcase the error: https://github.com/OpenCerts/opencerts-functions/commit/9f6bdd25b2268896f17c1f421f51f3a5dac440c3

In the meantime for this PR, will revert to using `opencerts-verify` so we don't have to manually build a `verify` instance.

**Pull request merged**